### PR TITLE
SSH/SCP support, Dockerfile startup script

### PR DESCRIPTION
**Pull request by drewjj**
# PR Details
## Description
This adds a startup script to the Dockerfile for the jpo-ode to allow for log offloading support while running the jpo-ode within a Kubernetes environment. 

The running of the jpo-ode application itself is identical to how it was before. This script starts up a SSH/SCP server to allow for log offloading. OpenRC and OpenSSH must be run in this procedural order on an Alpine image that is managed by Kubernetes.

## Motivation and Context

To allow for log offloading within a Kubernetes environment.

## How Has This Been Tested?

Within the CDOT GCP Kubernetes environment.

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
[ODE Contributing Guide](https://github.com/usdot-jpo-ode/jpo-ode/blob/bugfix/Pull_request_template/docs/contributing_guide.md) 
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: startup_jpoode.sh
Comment (dmccoystephenson) [inline startup_jpoode.sh:1]: Should this script be moved to the `scripts` directory?

File: Dockerfile [hunk 1]
```diff
@@ -23,15 +23,10 @@ WORKDIR /home
...
-	"-Dcom.sun.management.jmxremote.port=9090", \
-	"-Dcom.sun.management.jmxremote.rmi.port=9090", \
-	"-Dcom.sun.management.jmxremote", \
-	"-Dcom.sun.management.jmxremote.local.only=true", \
-	"-Dcom.sun.management.jmxremote.authenticate=false", \
-	"-Dcom.sun.management.jmxremote.ssl=false", \
-	"-Dlogback.configurationFile=/home/logback.xml", \
-	"-jar", \
-	"/home/jpo-ode-svcs.jar"]
+RUN apk add openssh
```
Comment (dmccoystephenson) [inline Dockerfile:28]: Do any additional ports need to be exposed for the SSH functionality to work? If so, these should be added to the 'ode' service in the docker compose files.

File: startup_jpoode.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+# NOTE: This is a bash script intended to be used by the Dockerfile and not intended to be used on its own
+
+# Create the necessary openrc directories on boot of the new container
+sh -c rc-status
+# Create the necessary file to run a service on openrc. This is necessary when using an alpine image
+touch /run/openrc/softlevel
+# Start the SSH/SCP server on boot of the new container
+rc-service sshd start
```
Comment (dmccoystephenson) [inline startup_jpoode.sh:10]: What password can be used to SSH into the container after spinning it up locally?

**Review by dmccoystephenson — CHANGES_REQUESTED** [review]

File: startup_jpoode.sh (hunk 2 above)
Comment (drewjj) [reply startup_jpoode.sh:10]: It shouldn't require one with a properly created SSH key that can be applied to the authorized_keys through either a local volume or an attached volume in K8s. I can specify this in the documentation.

File: Dockerfile (hunk 1 above)
Comment (drewjj) [reply Dockerfile:28]: Yeah I can expose this to work locally. I haven't tested it locally so I can't really vouch for a way to do that properly. This is definitely a K8s solution.

File: startup_jpoode.sh
Comment (drewjj) [reply startup_jpoode.sh:1]: I have moved this to the scripts directory.

File: Dockerfile (hunk 1 above)
Comment (drewjj) [reply Dockerfile:28]: I have thought about this for longer and I actually don't think I would recommend running this locally. This is really an inconvenient way to manage SSH access to the ODE system if you have the ability to use a standard host machine's SSH server. Mounting a directory from that host machine to the Dockerized ODE is definitely the best way to go about this. I have included the K8s documentation for this rationale and how to deploy it in K8s where we do expose the port.

File: startup_jpoode.sh (hunk 2 above)
Comment (drewjj) [reply startup_jpoode.sh]: I have added a new comment line to explain this a bit. I have also included how to add this authorized_keys through a mounted volume. Once again, though, I only recommend using this option in K8s. Having an unexposed SSH server isn't going to cause any issues, however.

File: Dockerfile (hunk 1 above)
Comment (dmccoystephenson) [reply Dockerfile:28]: Sounds good!

**Review by dmccoystephenson — APPROVED** [review]
Looks good to me!

**Review by mwodahl — APPROVED** [review]
Looks good to me!

**Pull request merged**